Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -402,69 +456,63 @@ void ConvertUnits::convertViaTOF(Kernel::Unit_const_sptr fromUnit,
           efixedProp = 0.0;
         }
       }
-    } else {
-      // set the Ei value in the run parameters
-      API::Run &run = outputWS->mutableRun();
-      run.addProperty<double>("Ei", efixedProp, true);
     }
   } else if (emode == 0 && efixedProp == EMPTY_DBL()) // Elastic
   {
     efixedProp = 0.0;
   }
 
   std::vector<std::string> parameters =
-      outputWS->getInstrument()->getStringParameter("show-signed-theta");
+      inputWS->getInstrument()->getStringParameter("show-signed-theta");
   bool bUseSignedVersion =
       (!parameters.empty()) &&
       find(parameters.begin(), parameters.end(), "Always") != parameters.end();
   function<double(const IDetector &)> thetaFunction =
       bUseSignedVersion
-          ? bind(&MatrixWorkspace::detectorSignedTwoTheta, outputWS, _1)
-          : bind(&MatrixWorkspace::detectorTwoTheta, outputWS, _1);
+          ? bind(&MatrixWorkspace::detectorSignedTwoTheta, inputWS, _1)
+          : bind(&MatrixWorkspace::detectorTwoTheta, inputWS, _1);
+
+  // Perform Sanity Validation before creating workspace
+  double checkefixed = efixedProp;
+  double checkl2;
+  double checktwoTheta;
+  size_t checkIndex = 0;
+  if (getDetectorValues(outputUnit, source, sample, l1, emode, inputWS,
+                        thetaFunction, checkIndex, checkefixed, checkl2,
+                        checktwoTheta)) {
+    const double checkdelta = 0.0;
+    // copy the X values for the check
+    auto checkXValues = inputWS->readX(checkIndex);
+    // Convert the input unit to time-of-flight
+    Unit *checkFromUnit = fromUnit->clone();
+    Unit *checkOutputUnit = outputUnit->clone();
```

Comment:
Could be simply dropped into a `std::unique_ptr`, then we can avoid the manual `deletes` below.